### ISSUE TITLE
Pipeline resources

### ISSUE THREAD

**Issue by MalbertR** [jtamames/SqueezeMeta#201]
I did the test run using this pipeline and after some start up problem with perl libraries it was able to finish successfully, which I'm really happy and excited about!
However, it did take a bit more time than I anticipated to complete. It took about 22 hours (21 hours, 48 minutes, 21 seconds to be exact), while given 100G memory and 4 threads. 
I am on a high performance computer with a lot of resources (up to 1TB memory, 48 threads and 2 weeks runtime, however I'm hoping I wouldn't need that much, since my job would be held in the queue for more than a couple of days probably, before finally starting), so obviously I would be able to crank these resources numbers up when I'm ready to run some real samples. Do you have any recommendation for the mount of resources (memory, threads and expected runtime) I should ask for, say a High-output NextSeq run (700-900 million reads in total)?

Thanks in advance!

**Comment by fpusan** [comment]
Hi!
Sorry for the late response. It is not surprising that the test run took that much if you only used 4 threads.
It is hard to estimate resource usage and runtime from the number of reads alone, as they will strongly depend on the complexity of the community you are analyzing. As a reference, it took us 14 days to run 14 marine samples (1167 million reads /142 gigabases in total) in a 750 Gb RAM server using 22 cores.

**Comment by MalbertR** [comment]
Thanks for the response.

Yes, I figured, the test consisted of only 2 samples so it shouldn't need much, but apparently it needed more than I thought. :D 

I have now been able to run the pipeline on 12 human fecal samples with an average of about 20 million reads each. I gave the job 250G RAM, 24 cores and it took 3 days and 19 hours to run (I got lucky, since the job would have been killed after 4 days), which is pretty acceptable (using default settings with `--cleaning`). I will probably be bumping the number of cores to 32 or maybe even 48 next time. 
Here is another finding...After the run, I checked on the max amount of RAM used and this was "only" about 86GB, so at least for Megahit there's not a whole lot of memory needed. I knew Megahit would need less memory compared to SPAdes, but I wasn't expecting this big of a difference, since I have used (meta)SPAdes before on these same samples and it needed over 500GB RAM to run (if I remember correctly). 

Anyway, I am really thankful for your efforts in developing/maintaining this pipeline and also helping me with these "issues". Looking forward to using it a whole lot more in the future :)

Closing this issue.
